### PR TITLE
Add option to move Tile/GridMap editors to another side

### PR DESCRIPTION
**Pull request by YeldhamDev**
Which the default for both being to the right, as the extra buttons in the top bar are to the right as well.

peek 2018-08-21 03-20

Also, I divided the EditorPlugin's `CONTAINER_*_EDITOR_SIDE` into left and right, to directly choose which side of the split to put controls.

### CODE REVIEW

**Comment by homer666** [comment]
Nice, having the panel on the right doesn't cause the 2D/3D view and its contents to shift whenever it appears or disappears. I don't know if that was the intention of this PR, but it's very much appreciated regardless!

**Comment by YeldhamDev** [comment]
@homer666 Seems to be how SplitContainers work.

**Comment by akien-mga** [comment]
I'm fine with having the possibility to change the position of that dock, but as we discussed with @reduz on IRC, not sure if it's good to have that as an editor setting. Editor settings for such things are cumbersome and most users never notice their existence.

It would be better, albeit more complex, to have a system to change the position as for the main editor docks.

If we stick to a setting, I think a "Left,Right" hint would be better than a boolean for "move to left".

CC @MarianoGnu @groud

**Comment by groud** [comment]
I agree with Akien, but maybe an editor setting is fine until we implement that?

**Comment by MarianoGnu** [comment]
I agree too, a button like the ones on the docks would do, this is fine for now imo, not's not too different than before

**Comment by YeldhamDev** [comment]
Left and right hints added.

File: editor/plugins/tile_map_editor_plugin.cpp [hunk 1]
```diff
@@ -1909,6 +1909,16 @@ TileMapEditor::~TileMapEditor() {
 ///////////////////////////////////////////////////////////////
 ///////////////////////////////////////////////////////////////
 
+void TileMapEditorPlugin::_notification(int p_what) {
+
+	if (p_what == EditorSettings::NOTIFICATION_EDITOR_SETTINGS_CHANGED) {
+		if (EditorSettings::get_singleton()->get("editors/tile_map/editor_side"))
```
Comment (akien-mga) [inline editor/plugins/tile_map_editor_plugin.cpp]: Please make this more explicit with a comment (and maybe a switch), e.g.:
```
switch ((int)EditorSettings::get_singleton()->get("editors/tile_map/editor_side")) {
	case 0: // Left
		CanvasItemEditor::get_singleton()->get_palette_split()->move_child(tile_map_editor, 0);
	case 1: // Right
		CanvasItemEditor::get_singleton()->get_palette_split()->move_child(tile_map_editor, 1);
```

**Comment by YeldhamDev** [comment]
Done and done!

File: editor/plugins/tile_map_editor_plugin.cpp [hunk 2]
```diff
@@ -1942,9 +1957,14 @@ TileMapEditorPlugin::TileMapEditorPlugin(EditorNode *p_node) {
 	EDITOR_DEF("editors/tile_map/sort_tiles_by_name", true);
 	EDITOR_DEF("editors/tile_map/bucket_fill_preview", true);
 	EDITOR_DEF("editors/tile_map/show_tile_info_on_hover", true);
+	EDITOR_DEF("editors/tile_map/editor_side", 1);
+	EditorSettings::get_singleton()->add_property_hint(PropertyInfo(Variant::INT, "editors/tile_map/editor_side", PROPERTY_HINT_ENUM, "Left,Right"));
 
 	tile_map_editor = memnew(TileMapEditor(p_node));
-	add_control_to_container(CONTAINER_CANVAS_EDITOR_SIDE, tile_map_editor);
+	if (EditorSettings::get_singleton()->get("editors/tile_map/editor_side"))
```
Comment (akien-mga) [inline editor/plugins/tile_map_editor_plugin.cpp]: Same comment here - if you find the switch case too much I'm ok with the `if`, but then I'd make it
`if (EditorSettings::get_singleton()->get("editors/tile_map/editor_side") == 1) { // Right`

Whichever you prefer, should be made consistent with the style used in `_notification`.

Comment (YeldhamDev) [reply editor/plugins/tile_map_editor_plugin.cpp]: Sorry 'bout that, completely forgot about that section. I'm fine with using `switch`.

**Comment by akien-mga** [comment]
BTW for future changes where you might need to set processing, make sure to use the *internal* process (`set_process_internal(true)` / `NOTIFICATION_INTERNAL_PROCESS`).

**Comment by akien-mga** [comment]
Thanks @YeldhamDev :)

**Comment by MarianoGnu** [comment]
> BTW for future changes where you might need to set processing, make sure to use the internal process (set_process_internal(true) / NOTIFICATION_INTERNAL_PROCESS).

Great, i didn't know of this one

**Pull request merged**